analyses/cell-type-ewings/exploratory_analysis/08-merged-celltypes.Rmd

```diff
@@ -0,0 +1,742 @@
+  t() |> 
+  as.data.frame()
+gene_cts$barcodes <- rownames(gene_cts)
+
+# get all unique cell types from the final lumped group 
+celltypes_df <- all_info_df |> 
+  dplyr::select(barcodes, final_lumped)
+
+# create a df that has gene expression column and column indicating whether or not the gene is detected in that cell 
+genes_df <- gene_cts |> 
+  tidyr::pivot_longer(!barcodes, names_to = "ensembl_gene_id", values_to = "gene_exp") |> 
+  dplyr::left_join(celltypes_df, by = c("barcodes")) |> 
+  dplyr::left_join(validation_markers_df, by = c("ensembl_gene_id")) |> 
+  dplyr::rowwise() |> 
+  dplyr::mutate(
+    detected = gene_exp > 0 # column indicating if gene is present or not
+  )
+
+# get total number of cells per final annotation group 
+total_cells_df <- genes_df |> 
+  dplyr::select(barcodes, final_lumped) |> 
+  unique() |> 
+  dplyr::count(final_lumped, name = "total_cells")
+
+# get total number of cells each gene is detected in per group 
+# and mean gene expression per group 
+group_stats_df <- genes_df |> 
+  dplyr::group_by(final_lumped, ensembl_gene_id) |>
+  dplyr::summarize(
+    detected_count = sum(detected),
+    mean_exp = mean(gene_exp)
+  )
+
+gene_summary_df <- genes_df |> 
+  # add total cells
+  dplyr::left_join(total_cells_df, by = c("final_lumped")) |> 
+  # add per gene stats
+  dplyr::left_join(group_stats_df, by = c("ensembl_gene_id", "final_lumped")) |> 
+  dplyr::select(gene_symbol, final_lumped, cell_type, total_cells, detected_count, mean_exp) |> 
+  unique() |>
+  dplyr::rowwise() |> 
+  dplyr::mutate(
+    # get total percent
+    percent_exp = (detected_count/total_cells) * 100,
+    # order genes based on cell type they indicate
+    gene_symbol = factor(gene_symbol, levels = validation_markers_df$gene_symbol),
+    final_lumped = forcats::fct_relevel(final_lumped, cell_type_order)
+    
+  ) 
+```
+
+```{r, fig.height=7, fig.width=10}
+# filter out low expressed genes
+dotplot_df <- gene_summary_df |> 
+  dplyr::filter(mean_exp > 0, percent_exp > 10) |>
+  dplyr::arrange(final_lumped) |> 
+  dplyr::mutate(y_label = as.factor(glue::glue("{final_lumped} ({total_cells})")))
+
+
+dotplot <- ggplot(dotplot_df, aes(y = forcats::fct_rev(y_label), x = gene_symbol, color = mean_exp, size = percent_exp)) +
+  geom_point() +
+  scale_color_viridis_c(option = "magma", limits = c(0,2.5), oob = scales::squish) +
+  theme(
+    axis.text.x = element_text(angle = 90, hjust = 0.5)
+  ) +
+  labs(
+    x = "",
+    y = "Final cell type annotation"
+  )
+
+# specify legend order 
+color_order <- dotplot_df |> 
+  dplyr::pull(cell_type) |> 
+  unique()
+
+color_bar <- ggplot(dotplot_df, aes(x = gene_symbol, y = 1, fill = cell_type)) + 
+  geom_tile() + 
+  scale_fill_brewer(palette = 'Set1', breaks = color_order) +
+  ggmap::theme_nothing() +
+  theme(legend.position = "bottom") +
+  labs(fill = "")
+
+dotplot + color_bar +
+  patchwork::plot_layout(ncol = 1, heights = c(4, 0.1)) 
+```
+
+A few notes from this plot: 
```

Comment:
I'd state that expression in the plot is capped at 2.5 but it might actually be higher